tuf/metadata/v1/tuf.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright 2018 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/GoogleCloudPlatform/runtimes-common/tuf/config"
+	"github.com/GoogleCloudPlatform/runtimes-common/tuf/constants"
+	"github.com/GoogleCloudPlatform/runtimes-common/tuf/metadata"
+	"github.com/GoogleCloudPlatform/runtimes-common/tuf/types"
+)
+
+var HASHALGOS = []types.HashAlgo{
+	types.HashAlgo("sha256"),
+}
+
+type TUF struct {
+	TufConfig       config.TUFConfig
+	RootSecrets     []types.Scheme
+	OldRootSecrets  []types.Scheme
+	TargetSecrets   []types.Scheme
+	SnapshotSecrets []types.Scheme
+}
+
+func (tuf *TUF) PopulateRootMetadata() (RootMetadata, error) {
```

Comment:
It's hard to see in this diff how these populate methods will be used. Depending on the usage, it might make more sense to do this automatically during TUF initialization. WDYT?

Reply:
Yes. Sorry, i shd have this clear in the PR description.
These will be used when someone uploads a new key using upload_secrets command.
The TODO here will be replaced by these functions. Based on which new keys are uploaded, we will re-generate all or few metadata files and upload them back to gcs.
https://github.com/GoogleCloudPlatform/runtimes-common/blob/master/tuf/deployer/deployer.go#L83.

Reply:
Forgot to add, you can't do this in init because you dont have all the secrets.